src/System.Private.Xml/src/System/Xml/Serialization/CodeGenerator.cs

```diff
@@ -1418,7 +1418,15 @@ public WhileState(CodeGenerator ilg)
         internal void WhileBegin()
         {
             WhileState whileState = new WhileState(this);
-            Br(whileState.CondLabel);
+            // ECMA-335 III.1.7.5 states that code blocks after unconditional
```

Comment:
Note that .NET Framework and .NET Core JITs never took advantage of this restriction and always handled backward branches to non-empty stacks just fine. There is a lot of code out there with backward branches to non-empty stacks.

Reply:
If we ever get to updating the spec, we are likely to delete this restriction from it.

Reply:
Mono can handle it in certain modes (eg. interpreter) but not in others (the default JIT). The restriction concerns only backward branches to non-empty stack where no forward branch to the same code exists. The C# compiler never generates this AFAIK.

If you consider to be something that should be supported then we should raise an issue with Mono to allow it.

Reply:
@jkotas do you think this is something Mono will have to implement for .net 5? This does not look like something used often as I don't remember seeing this before.

Reply:
@marek-safar I don't think it would be particularly hard to implement if you decide that's the right way to go. I can even give it a try.

I do expect `XmlSerializer` to be quite important to get right in .NET 5 but whether it is accomplished by changing Mono or CoreFX I am impartial on.

Reply:
> is this is something Mono will have to implement for .net 5? 

Depends on whether something else runs into it, how hard the fix would be and how much time is spent diagnosing the problem (the xml serializer took like 4 people to diagnose). 

The proposed fix in the serializer does not look too distruptive, so I would not have a problem with taking it if you prefer that.

IIRC, .NET Native took advantage of this restriction originally, but then was fixed to match .NET Framework because of somebody reported it as compat problem.

Reply:
I did search through Mono GitHub issues to look for reports of  issues with the same root cause. Aside from the ones triggering this investigation there was one (https://github.com/mono/mono/issues/8077) reported to be a code generated by obfuscator. All other "invalid IL" issues were caused by different bugs, mostly thread-safety issue in dynamic code generation that was recently fixed.

Reply:
Further search shows that it was hit back in 2004 by IKVM: http://weblog.ikvm.net/2004/03/29/BackwardBranchConstraints.aspx.